File: app/src/main/java/com/esbati/keivan/persiancalendar/features/settings/cells/TextInfoCell.java
/*
 * This is the source code of Telegram for Android v. 3.x.x.
 * It is licensed under GNU GPL v. 2 or later.
 * You should have received a copy of the license in this archive (see LICENSE).
 *
 * Copyright Nikolai Kudashov, 2013-2016.
 */

package com.esbati.keivan.persiancalendar.features.settings.cells;

import android.content.Context;
import android.util.TypedValue;
import android.view.Gravity;
import android.widget.FrameLayout;
import android.widget.TextView;

import com.esbati.keivan.persiancalendar.utils.LayoutHelper;

import static com.esbati.keivan.persiancalendar.utils.AndroidUtilitiesKt.toDp;

public class TextInfoCell extends FrameLayout {

    private TextView textView;

    public TextInfoCell(Context context) {
        super(context);

        textView = new TextView(context);
        textView.setTextColor(0xffa3a3a3);
        textView.setTextSize(TypedValue.COMPLEX_UNIT_DIP, 13);
        textView.setGravity(Gravity.CENTER);
        textView.setPadding(0, toDp(19), 0, toDp(19));
        addView(textView, LayoutHelper.createFrame(LayoutHelper.WRAP_CONTENT, LayoutHelper.WRAP_CONTENT, Gravity.CENTER, 17, 0, 17, 0));
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED));
    }

    public void setText(String text) {
        textView.setText(text);
    }

    public void setTextColor(int textColor){
        textView.setTextColor(textColor);
    }
}

File: app/src/main/java/com/esbati/keivan/persiancalendar/features/settings/cells/TextInfoCell.kt
package com.esbati.keivan.persiancalendar.features.settings.cells

import android.content.Context
import android.util.TypedValue
import android.view.Gravity
import android.view.View
import android.widget.FrameLayout
import android.widget.TextView
import com.esbati.keivan.persiancalendar.features.settings.ViewMarker
import com.esbati.keivan.persiancalendar.utils.LayoutHelper
import com.esbati.keivan.persiancalendar.utils.toDp

@ViewMarker
class TextInfoCell(context: Context) : FrameLayout(context) {

    private val textView: TextView
    var text: String
        get() = textView.text.toString()
        set(value) {
            textView.text = value
        }
    var textColor: Int = -0x5c5c5d
        set(value) {
            field = value
            textView.setTextColor(value)
        }

    init {
        textView = TextView(context).apply {
            setTextColor(textColor)
            setTextSize(TypedValue.COMPLEX_UNIT_DIP, 13f)
            gravity = Gravity.CENTER
            setPadding(0, 19.toDp(), 0, 19.toDp())
        }
        addView(textView, LayoutHelper.createFrame(
                LayoutHelper.WRAP_CONTENT
                , LayoutHelper.WRAP_CONTENT.toFloat()
                , Gravity.CENTER
                , 17f, 0f, 17f, 0f)
        )
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED))
    }
}
